server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/symbols/SymbolAccumulator.java

```diff
@@ -147,11 +148,35 @@ public Optional<SyntaxError> registerCodeBlockUsage(
     CodeBlockDefinitionNode definition = definitions.get(0);
     definition.addUsage(node.getLocality());
 
-    Optional.ofNullable(symbolTable.getParagraphMap().get(node.getName()))
-        .ifPresent(it -> it.addUsage(node.getLocality().toLocation()));
-    Optional.ofNullable(symbolTable.getSectionMap().get(node.getName()))
-        .ifPresent(it -> it.addUsage(node.getLocality().toLocation()));
+    Location location = node.getLocality().toLocation();
+    String definitionName = definition.getName();
 
+    // Try to resolve exact procedure identification
+    // 1. check if we are in a section
+    String trySameSection = definition.getNearestParentByType(NodeType.PROCEDURE_SECTION)
+            .map(n -> ((ProcedureSectionNode) n).getName())
+            .orElse(null);
+    // If there is paragraph with the same name - call it
```

Comment:
Wasn't this exact thing been done in the if statement above?

Reply:
Ideally if statement above should be always false in correct COBOL code. But I plan to work on it on the next iteration.